### PR TITLE
AB Test | Single Newsletter Test

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?
Add AB Test: `SingleNewsletterTest`

The AB test shows a single newsletter (Today UK/US/AU dependant on region) on the consents newsletter page for the `variant`, while showing 4 newsletters in the control. The hypothesis being that showing a single newsletter, vs 4, will increase opt in to at least one newsletter.

Also in the `variant` on the first page of the consents flow (`/consents/communication`) we hide the market research opt out consent checkbox/information, vs the `control` where we show this checkbox.

## Changes

- Add the `SingleNewsletterTest`
- Cleanup the `EnhancedConsentTest`
- Roll out the updated header from the `EnhancedConsentTest`
- General cleanup e.g. move shared code into single variable/method etc.
- Update dependencies

## Images

Variant only

#### Desktop
profile code dev-theguardian com_consents_newsletters_returnUrl=https%3A%2F%2Fm code dev-theguardian com%2F(Laptop with HiDPI screen)

#### Mobile
profile code dev-theguardian com_consents_newsletters_returnUrl=https%3A%2F%2Fm code dev-theguardian com%2F(iPhone 6_7_8)

#### Desktop
profile code dev-theguardian com_consents_communication_returnUrl=https%3A%2F%2Fm code dev-theguardian com%2F(Laptop with HiDPI screen)

#### Mobile
profile code dev-theguardian com_consents_communication_returnUrl=https%3A%2F%2Fm code dev-theguardian com(iPhone 6_7_8)

#### Desktop
profile code dev-theguardian com_consents_review_returnUrl=https%3A%2F%2Fm code dev-theguardian com%2F(Laptop with HiDPI screen)

#### Mobile
profile code dev-theguardian com_consents_review_returnUrl=https%3A%2F%2Fm code dev-theguardian com%2F(iPhone 6_7_8)

## Tested
- [x] CI
- [x] CODE

### CODE REVIEW

File: cypress/integration/onboarding_flow.spec.js [hunk 1]
```diff
@@ -772,4 +772,134 @@ describe('Onboarding flow', () => {
       cy.contains('the page does not exist');
     });
   });
+
+  context('AB Test - Single Newsletter test', () => {
```
Comment (coldlink) [inline cypress/integration/onboarding_flow.spec.js:776]: cypress tests for the single newsletter test

File: cypress/support/commands.js [hunk 2]
```diff
@@ -71,3 +71,9 @@ Cypress.Commands.add('idapiLastPayloadIs', (expected) => {
 Cypress.Commands.add('idapiMockPurge', () => {
   cy.request(MOCKING_ENDPOINT + '/purge');
 });
+
+Cypress.Commands.add('setMvtId', (str) => {
```
Comment (coldlink) [inline cypress/support/commands.js:75]: adds a cypress command so we can force ourselves into certain variants

File: src/client/components/NewsletterCardAB.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
+import React, { FunctionComponent } from 'react';
```
Comment (coldlink) [inline src/client/components/NewsletterCardAB.tsx:1]: this is the complete component for the single newsletter "card" for the variant, based on the original newsletter card at `src/client/components/NewsletterCard.tsx`

File: src/client/layouts/shared/Consents.tsx [hunk 4]
```diff
@@ -1,23 +1,21 @@
 import React, { FunctionComponent } from 'react';
```
Comment (coldlink) [inline src/client/layouts/shared/Consents.tsx:1]: the changes in this file are from cleaning up the `EnhancedConsentTest` code, and adding in shared code/new header too

File: src/client/pages/ConsentsConfirmation.tsx [hunk 5]
```diff
@@ -184,11 +183,13 @@ export const ConsentsConfirmationPage = () => {
                 <p css={text}>N/A</p>
               )}
             </ReviewTableRow>
-            <ReviewTableRow title="Marketing research">
-              <p css={text}>
-                {market_research_optout.consented ? 'No' : 'Yes'}
-              </p>
-            </ReviewTableRow>
+            {!isUserInTest && (
```
Comment (coldlink) [inline src/client/pages/ConsentsConfirmation.tsx:186]: hides the market research optout from the variant

**Review by rebecca-thompson — APPROVED** [review]

File: src/shared/model/experiments/tests/singleNewsletterTest.ts [hunk 6]
```diff
@@ -1,17 +1,18 @@
...
+export const singleNewsletterTest: ABTest = {
+  id: 'SingleNewsletterTest', // This ID must match the Server Side AB Test
+  start: '2021-04-09',
+  expiry: '2021-05-09', // Remember that the server side test expiry can be different
+  author: 'mahesh.makani@theguardian.com',
   description:
-    'Testing the effect of having an enhanced consent design and a simplified header on user consents',
+    'We believe that if we show a single newsletter, vs 4, it will increase opt in to at least one newsletter. . We think it will increase by at least 7.5%',
   audience: 1, // 0.01% (1 is 100%)
   audienceOffset: 0, // 50% (1 is 100%). Prevent overlapping with other tests.
```
[the remark this answers is not in the document]
Comment (coldlink) [reply src/shared/model/experiments/tests/singleNewsletterTest.ts:11]: ah woops! the file was copied over from the example file, I'll remove the comments about 0.01% and 50%

**Pull request merged**